python_modules/dagster/dagster/_core/telemetry.py

```diff
@@ -538,6 +545,49 @@ def get_stats_from_external_repo(external_repo: "ExternalRepository") -> Mapping
     }
 
 
+def get_resource_stats(external_resources: Sequence["ExternalResource"]) -> Mapping[str, str]:
+    dagster_resources = [
+        "dagster_snowflake.resources.SnowflakeResource",
+        "dagster_dbt.cli.resources.DbtCliClientResource",
+        "dagster_snowflake.resources.snowflake_resource"
+        # fill out with rest of resources
+    ]
+
+    dagster_io_managers = [
+        "dagster_snowflake_pandas.snowflake_pandas_type_handler.SnowflakePandasIOManager",
+        "dagster_snowflake_pandas.snowflake_pandas_type_handler.snowflake_pandas_io_manager",
+        "dagster_snowflake_pyspark.snowflake_pyspark_type_handler.SnowflakePySparkIOManager",
+        "dagster_snowflake_pyspark.snowflake_pyspark_type_handler.snowflake_pyspark_io_manager",
+        "dagster_snowflake.snowflake_io_manager.snowflake_io_manager"
+        # fill out with rest of io managers
+    ]
+
+    used_dagster_resources = []
+    used_dagster_io_managers = []
+    used_custom_resources = []
+    # used_custom_io_managers = (
+    #     []
+    # )  # still figuring out how to distinguish custom resources and io managers, so just doing resources for now
+
+    for resource in external_resources:
+        resource_type = resource.resource_type
+        if resource_type in dagster_resources:
+            used_dagster_resources.append(resource_type)
+        elif resource_type in dagster_io_managers:
+            used_dagster_io_managers.append(resource_type)
+        else:
+            used_custom_resources.append(resource_type)
+
+    d = {
+        "dagster_resources": str(used_dagster_resources),
+        "dagster_io_managers": str(used_dagster_io_managers),
+        "custom_resources": str(used_custom_resources),
+        # "custom_io_managers": str(used_custom_io_managers),
+    }
```

Comment:
i wonder if we could just send raw blob per line like `{"resource_name": "dagster_snowflake_pandas.snowflake_pandas_type_handler.SnowflakePandasIOManager"}` and aggregate it in post-processing in a purina job. so that if anything we can post-process or backfill as opposed to updating telemetry.

Reply:
yeah that makes a lot of sense

Reply:
I like the idea of nesting the resources so that the top level in the hierarchy doesn't get to crowded

if we do this, we should avoid including custom IO managers in the blob, because I think that would be an invasion of users' privacy

I also think we should keep this resilient to refactors of our integration libraries. i.e. I think the names should use what gets imported by the user, not include internal packages in the path.

Reply:
> if we do this, we should avoid including custom IO managers in the blob, because I think that would be an invasion of users' privacy

makes sense, do you think having a boolean indicating the presence of custom resources or io managers would be ok? that seems like useful info to know 

> I also think we should keep this resilient to refactors of our integration libraries. i.e. I think the names should use what gets imported by the user, not include internal packages in the path.

that also makes sense. Right now the info we get in telemetry is the full class path, but i could cut out the middle bit and just keep the module (ie `dagster-snowflake-pandas`) and the class name (ie `SnowflakePandasIOManager`)

Reply:
ok so in sum it seems like we want to:
1. record the module and class name of all dagster resources and IO managers that are used
2. potentially have a boolean indicating if there are custom resources/io managers 

The main difficulty with 1 will be determining which resources are dagster's vs which ones are the users

Reply:
> makes sense, do you think having a boolean indicating the presence of custom resources or io managers would be ok? that seems like useful info to know

yeah, definitely